react/features/authentication/middleware.ts

```diff
@@ -100,26 +106,72 @@ MiddlewareRegistry.register(store => next => action => {
         if (_isWaitingForOwner(store)) {
             store.dispatch(stopWaitForOwner());
         }
-        _hideLoginDialog(store);
+        store.dispatch(hideLoginDialog());
         break;
 
     case CONFERENCE_LEFT:
         store.dispatch(stopWaitForOwner());
         break;
 
     case CONNECTION_ESTABLISHED:
-        _hideLoginDialog(store);
+        store.dispatch(hideLoginDialog());
         break;
 
     case CONNECTION_FAILED: {
         const { error } = action;
+        const state = store.getState();
+        const { jwt } = state['features/base/jwt'];
 
         if (error
                 && error.name === JitsiConnectionErrors.PASSWORD_REQUIRED
-                && typeof error.recoverable === 'undefined') {
+                && typeof error.recoverable === 'undefined'
+                && !jwt) {
             error.recoverable = true;
+
+            const config = state['features/base/config'];
+            const room = getBackendSafeRoomName(state['features/base/conference'].room);
+
+            // this is only for web
+            if (isTokenAuthEnabled(config) && typeof APP !== 'undefined') {
+                // FIXME: This method will not preserve the other URL params that were originally passed.
+                // redirectToTokenAuthService
+                window.location.href = getTokenAuthUrl(config)(room, false);
+            } else {
+                store.dispatch(openLoginDialog());
```

Comment:
Shouldn't we should a notification and go back to the welcome page if there was no token but `isTokenAuthEnabled(config)` is set?

Reply:
When there is no token set and and we have the token URL we redirect to the token URL, what do you mean?

Reply:
Like this, right? 
https://github.com/jitsi/jitsi-meet/pull/13538/commits/9d62f6785c369ab87af2d69a276bf665e685a570

Reply:
Yes!